gapic/src/main/com/google/gapid/models/Settings.java

```diff
@@ -50,133 +56,209 @@
  */
 public class Settings {
   private static final Logger LOG = Logger.getLogger(Settings.class.getName());
+
   private static final String SETTINGS_FILE = ".gapic";
   private static final int MAX_RECENT_FILES = 16;
+  private static final int CURRENT_VERSION = 0;
```

Comment:
Curious... What's this version number used for? Didn't find its usage after a quick search...

Reply:
The idea here is to allow non-0/false/"" default values. When adding a new field that has a different default, the current version can be incremented and in the `fixup` function we can test for older versions and set the default. E.g. say current version is v0, I add a field that I want to default to "foo". I increment the version to v1 and in `fixup` if the version of the deserialized proto is v0 or earlier, we can just set the property to "foo", since it will not be present.

It's not yet used, since we are at the first version :)